target_postgres/sinks.py

```diff
@@ -252,3 +255,62 @@ def schema_name(self) -> Optional[str]:
 
         # Schema name not detected.
         return None
+
+    def activate_version(self, new_version: int) -> None:
+        """Bump the active version of the target table.
+
+        Args:
+            new_version: The version number to activate.
+        """
+        # There's nothing to do if the table doesn't exist yet
+        # (which it won't the first time the stream is processed)
+        if not self.connector.table_exists(self.full_table_name):
+            return
+
+        deleted_at = now()
+        datetime_type = self.connector.to_sql_type(
+            {"type": "string", "format": "date-time"}
+        )  # Different from SingerSDK as we need to handle types the same as SCHEMA messsages
+        integer_type = self.connector.to_sql_type(
+            {"type": "integer"}
+        )  # Different from SingerSDK as we need to handle types the same as SCHEMA messsages
+
+        if not self.connector.column_exists(
+            full_table_name=self.full_table_name,
+            column_name=self.version_column_name,
+        ):
+            self.connector.prepare_column(
+                self.full_table_name,
+                self.version_column_name,
+                sql_type=integer_type,  # Different from SingerSDK as we need to handle types the same as SCHEMA messsages
+            )
+
+        self.logger.info("Hard delete: %s", self.config.get("hard_delete"))
+        if self.config["hard_delete"] is True:
+            self.connection.execute(
+                f"DELETE FROM {self.full_table_name} "
+                f"WHERE {self.version_column_name} <= {new_version} OR {self.version_column_name} IS NULL"
```

Comment:
I'm a little worried about what would happen if someone ran this in production with `add_record_metadata` set to True for a while, then suddenly they change it to `False`. 

I think we'd end up dropping the entire table, because we'd add all the new records, then none of them would have a `new_version` set they'd all be NULL. Then we'd drop all the NULLs.  :O 

Removing the IS NULL check would work here but I think the better thing to do would be a check to be sure metadata_columns is enabled as I think the IS NULL check is very valid here as in normal operation you want activate_version to be keeping your data very "clean" and in the same shape as the source. 

I should also write a test for this case as it's potentially scary (only place in our code that we drop tables so being cautious is good)

Reply:
If we fix the other issue https://github.com/MeltanoLabs/target-postgres/pull/89/files#r1084618420 I believe this one gets solved as well.